packages/python/plotly/plotly/express/_core.py

```diff
@@ -2052,36 +2291,35 @@ def get_groups_and_orders(args, grouper):
     of a single dimension-group
     """
     orders = {} if "category_orders" not in args else args["category_orders"].copy()
-
+    df: nw.DataFrame = args["data_frame"]
     # figure out orders and what the single group name would be if there were one
     single_group_name = []
     unique_cache = dict()
-    for col in grouper:
+    grp_to_idx = dict()
+
+    for i, col in enumerate(grouper):
         if col == one_group:
             single_group_name.append("")
         else:
             if col not in unique_cache:
-                unique_cache[col] = list(args["data_frame"][col].unique())
+                unique_cache[col] = df.get_column(col).unique().to_list()
             uniques = unique_cache[col]
             if len(uniques) == 1:
                 single_group_name.append(uniques[0])
             if col not in orders:
                 orders[col] = uniques
             else:
                 orders[col] = list(OrderedDict.fromkeys(list(orders[col]) + uniques))
-    df = args["data_frame"]
+
+    grp_to_idx = {k: i for i, k in enumerate(orders)}
```

Comment:
I am using this to replicate the final output that the rest of plotly uses but avoids having duplicate keys in group by, for which pandas has no issues but other backend raise an error